.github/workflows/ci-cd.yml

```diff
@@ -0,0 +1,77 @@
+name: CI/CD Pipeline
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+permissions:
+  contents: read
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+
+    services:
+      mysql:
+        image: mysql:5.7
+        ports:
+          - 3306:3306
+        env:
+          MYSQL_ROOT_PASSWORD: root
+          MYSQL_DATABASE: testdb
+          MYSQL_USER: testuser
+          MYSQL_PASSWORD: testpassword
+        options: >-
+          --health-cmd="mysqladmin ping --silent"
+          --health-interval=10s
+          --health-timeout=5s
+          --health-retries=3
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup JDK 21
+        uses: actions/setup-java@v3
+        with:
+          java-version: '21'
+          distribution: 'temurin'
+          cache: maven
+
+      - name: Build with Maven
+        run: mvn clean install -DskipTests
+
+      - name: Run Tests
+        run: mvn test
+
+      - name: Build Docker image
+        run: docker build . --file Dockerfile --tag purrgent-care:$(date +%s)
+
+      - name: Start Spring Boot application
+        run: nohup mvn spring-boot:run & echo $! > spring-boot.pid
+
+      - name: Wait for Spring Boot to start
+        run: |
+          echo "Waiting for Spring Boot to start..."
+          timeout 60s bash -c 'until curl -s http://localhost:8080/actuator/health | grep "UP"; do sleep 2; done'
+
+      - name: Set up Node.js
+        uses: actions/setup-node@v3
+        with:
+          node-version: '22'
+
+      - name: Install Newman
+        run: npm install -g newman
+
+      - name: Run Postman Collection
+        env:
+          PERSONS_BASE_URL: http://localhost:8080/persons
+          PET_BASE_URL: http://localhost:8080/pets
+          ACCOUNT_BASE_URL: http://localhost:8080/accounts
```

Comment:
These would be better as secrets in the repository but a repo admin needs to set those up.

Reply:
This will be addressed by this [issue](https://github.com/JesseMattson/PurrgentCare/issues/62) by @JesseMattson  😄